src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2151,541 +2181,160 @@ static void batch_upload_command(int connFd, bool *loop_exit_p) {
-
-    Utils::unzipFile(fullFilePath);
-    size_t lastindex = fileName.find_last_of(".");
-    string rawname = fileName.substr(0, lastindex);
-    fullFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + rawname;
+    instance_logger.info("Sent : " + JasmineGraphInstanceProtocol::OK);
 
-    while (!Utils::fileExists(fullFilePath)) {
-        line = Utils::read_str_trim_wrapper(connFd, data, INSTANCE_DATA_LENGTH);
-        if (line.compare(JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK) != 0) {
-            instance_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK +
-                                  " ; Received: " + line);
-            close(connFd);
-            return;
-        }
-        instance_logger.info("Received : " + line);
-        if (!Utils::send_str_wrapper(connFd, JasmineGraphInstanceProtocol::BATCH_UPLOAD_WAIT)) {
-            *loop_exit_p = true;
-            return;
-        }
-        instance_logger.info("Sent : " + JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK);
-    }
+    char data[DATA_BUFFER_SIZE];
+    string graphID = Utils::read_str_trim_wrapper(connFd, data, INSTANCE_DATA_LENGTH);
+    instance_logger.info("Received Graph ID: " + graphID);
 
-    line = Utils::read_str_wrapper(connFd, data, INSTANCE_DATA_LENGTH, false);
-    if (line.compare(JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK) != 0) {
-        instance_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::BATCH_UPLOAD_CHK +
-                              " ; Received: " + line);
-        close(connFd);
+    if (!Utils::send_str_wrapper(connFd, JasmineGraphInstanceProtocol::OK)) {
+        *loop_exit_p = true;
         return;
     }
-    instance_logger.info("Received : " + line);
-    if (!Utils::send_str_wrapper(connFd, JasmineGraphInstanceProtocol::BATCH_UPLOAD_ACK)) {
+    instance_logger.info("Sent : " + JasmineGraphInstanceProtocol::OK);
+
+    string partitionID = Utils::read_str_trim_wrapper(connFd, data, INSTANCE_DATA_LENGTH);
+    instance_logger.info("Received Partition ID: " + partitionID);
+
+    if (!Utils::send_str_wrapper(connFd, JasmineGraphInstanceProtocol::OK)) {
         *loop_exit_p = true;
         return;
     }
-    instance_logger.info("Sent : " + JasmineGraphInstanceProtocol::BATCH_UPLOAD_ACK);
+    instance_logger.info("Sent : " + JasmineGraphInstanceProtocol::OK);
+
+    string workerList = Utils::read_str_trim_wrapper(connFd, data, INSTANCE_DATA_LENGTH);
+    instance_logger.info("Received Worker List " + workerList);
+
+    std::vector<string> workerSockets;
+    stringstream wl(workerList);
+    string intermediate;
+    while (getline(wl, intermediate, ',')) {
+        workerSockets.push_back(intermediate);
+    }
+
+    JasmineGraphInstanceService::duplicateCentralStore(serverPort, stoi(graphID), stoi(partitionID), workerSockets,
+                                                       "localhost");
```

Comment:
We are sending `localhost` here. Is this acceptable? Can't there be scenarios where we need to use other host other than `localhost`?

Reply:
Done

Reply:
Ok